zebra-network/src/protocol/internal/response_status.rs

```diff
@@ -0,0 +1,101 @@
+//! The status of a response to an inventory request.
+
+use std::fmt;
+
+#[cfg(any(test, feature = "proptest-impl"))]
+use proptest_derive::Arbitrary;
+
+use ResponseStatus::*;
+
+/// A generic peer inventory response status.
+///
+/// `Available` is used for inventory that is present in the response,
+/// and `Missing` is used for inventory that is missing from the response.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+pub enum ResponseStatus<A, M> {
+    /// An available inventory item.
+    Available(A),
+
+    /// A missing inventory item.
+    Missing(M),
+}
+
+impl<A, M> fmt::Display for ResponseStatus<A, M> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(self.command())
+    }
+}
+
+impl<A, M> ResponseStatus<A, M> {
+    /// Returns the response status type as a string.
+    pub fn command(&self) -> &'static str {
+        match self {
+            ResponseStatus::Available(_) => "Available",
+            ResponseStatus::Missing(_) => "Missing",
+        }
+    }
+
+    /// Returns true if the inventory item was available.
+    #[allow(dead_code)]
```

Comment:
Minor: This could be removed.

```suggestion
```

Reply:
Fixed in 0c5f18ddf, along with all the other similar issues